adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java

```diff
@@ -0,0 +1,133 @@
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.hydrator.common.HiveSchemaConverter;
+import co.cask.hydrator.common.RecordConverter;
+import org.apache.hadoop.io.BooleanWritable;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.io.DoubleWritable;
+import org.apache.hadoop.io.FloatWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.orc.TypeDescription;
+import org.apache.orc.mapred.OrcStruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Creates ORCStruct records from StructuredRecords
+ */
+public class StructuredToOrcTransformer extends AbstractStructuredRecordTransformer<OrcStruct> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StructuredToOrcTransformer.class);
+  private final Map<Schema, TypeDescription> schemaCache = new HashMap<>();
+
+  public StructuredToOrcTransformer(Schema outputSchema) {
+    super(outputSchema);
+  }
+
+  @Override
+  public OrcStruct transform(StructuredRecord input, Schema schema) {
+    List<Schema.Field> fields = input.getSchema().getFields();
+    OrcStruct orcRecord = parseOrcSchema(input.getSchema());
+    // populate ORC struct orcRecord object
+    for (int i = 0; i < fields.size(); i++) {
+      Schema.Field field = fields.get(i);
+      try {
+        WritableComparable writable = convertToWritable(field, input);
+        orcRecord.setFieldValue(field.getName(), writable);
+      } catch (UnsupportedTypeException e) {
+        LOG.debug("{} is not a supported type", field.getName(), e);
+        throw new IllegalArgumentException(String.format("%s is not a supported type", field.getName()), e);
+      }
+    }
+    return orcRecord;
+  }
+
+  private OrcStruct parseOrcSchema(Schema inputSchema) {
+    OrcStruct orcRecord;
+    if (schemaCache.containsKey(inputSchema)) {
+      TypeDescription schema = schemaCache.get(inputSchema);
+      orcRecord = (OrcStruct) OrcStruct.createValue(schema);
+    } else {
+      StringBuilder builder = new StringBuilder();
+      try {
+        HiveSchemaConverter.appendType(builder, inputSchema);
+      } catch (UnsupportedTypeException e) {
+        throw new IllegalArgumentException(String.format("Not a valid Schema %s", inputSchema.toString()), e);
+      }
+      TypeDescription schema = TypeDescription.fromString(builder.toString());
+      orcRecord = (OrcStruct) OrcStruct.createValue(schema);
+      schemaCache.put(inputSchema, schema);
+    }
+    return orcRecord;
+  }
+
+  private WritableComparable convertToWritable(Schema.Field field, StructuredRecord input)
```

Comment:
this logic in this method seems quite similar to other code in hydrator-plugins repo

Reply:
Yeah, this is from https://github.com/caskdata/hydrator-plugins/blob/develop/core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToOrcTransformer.java

Reply:
I believe this is copied in other plugin projects as well. Can you open a jira to move common orc functions to some common lib module that all these projects can add as a dependency.

Reply:
Created the JIRA for the improvement: https://issues.cask.co/browse/CDAP-13076